pyccl/core.py

```diff
@@ -634,9 +661,29 @@ def compute_distances(self):
         """Compute the distance splines."""
         if self.has_distances:
             return
-        status = 0
-        status = lib.cosmology_compute_distances(self.cosmo, status)
-        check(status, self)
+        if not self.background_on_input:
+            status = 0
+            status = lib.cosmology_compute_distances(self.cosmo, status)
+            check(status, self)
+        else:
+            # Check that input arrays have the same size.
+            if not (self.a_array.shape == self.chi_a.shape == self.hoh0.shape):
```

Comment:
We should create a few quick unit tests for all of these ifs (i.e. tests that check that the corresponding errors are raised). I can help do that, ping me on slack.